### DISCUSSION TITLE
Splitting Up deskTool Plugin

### DISCUSSION THREAD

**Discussion by 3200creative** [sanity-io/sanity#3851] [Sanity Studio Support]

**Comment by 3200pro** [comment]
Hello:)

I've got a desk structure that should be split into multiple files due to its complexity.

Is there a public example or Gist of this? I'm having trouble creating `S.listItem` within files outside of `sanity.config.ts`

**Reply by 3200pro** [reply]
Feel free to delete this discussion (or show me how to). I had a typo that eluded me for a bit.

**Reply by mountagency** [reply]
@3200pro Did you ever find a solution for this issue?
We're currently experiencing the same.

Previously we have split up the desk structure into individual files for each document schemaType. and imported them from `../desk/*.ts` to our deskStructure.js.

Doing this in with the new V3 methods referenced [here](https://beta.sanity.io/docs/platform/studio/v2-to-v3/structure), results in error `expected linkType but received function`.

**Reply by 3200pro** [reply]
@mountagency 

I submitted this ticket and put it on hold until I hear back. Until then I'm keeping everything in the config.

It's nice to hear at least one other person ran into the same error. @rexxars or @kmelve  , Any advice or docs on how to set this up correctly?

**Reply by benbenpetit** [reply]
Hello,
I was struggling as you were, but I discover this Sanity x Shopify repo example that has been updated with v3.
https://github.com/sanity-io/sanity-shopify-studio

Enjoy 😊

**Reply by 3200pro** [reply]
Nice! Thanks for the heads up. This is 100% what I was looking for.

**Reply by benbenpetit** [reply]
Now you gotta parse StructureBuilder and Context on every child function. I don't find this really intuitive but it works !

**Reply by 3200pro** [reply]
I think that's what I was getting stuck on.
